src/aead.rs

```diff
@@ -664,6 +664,76 @@ enum Direction {
     Sealing,
 }
 
+trait BitXor {
+    fn bitxor_assign(&mut self, input: [u8; BLOCK_LEN]);
```

Comment:
Isn't this trait the same as `core::ops::BitXorAssign<[u8; BLOCK_LEN]`?

Reply:
It very likely is, can double check, and just use `core::ops::BitXorAssign<[u8; BLOCK_LEN]>` if possible. Will report any blockers. Was basically just re-implementing functions from `Block` as needed.